pkg/virtctl/create/vm.go

```diff
@@ -0,0 +1,450 @@
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-pvc=my-pvc
+
+  # Create a VirtualMachine with a specified Instancetype, Preference and DataVolume
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-dv=my-dv
+
+  # Create a VirtualMachine with a specified Instancetype, Preference and cloned PVC
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-clone-pvc=my-ns/my-pvc
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-dvt-pvc=my-ns/my-pvc
+
+  # Create a VirtualMachine with a specified Instancetype, Preference and cloned DataSource
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-clone-ds=my-ds
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-dvt-ds=my-ds
+
+  # Create a VirtualMachine with a specified Instancetype, Preference and cloned DataSource in namespace
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-clone-ds=my-ns/my-ds
+  {{ProgramName}} create vm --instancetype=my-instancetype --preference=my-preference --volume-dvt-ds=my-ns/my-ds`
+}
+
+func splitPrefixedName(prefixedName string) (prefix string, name string, err error) {
+	split := strings.Split(prefixedName, "/")
+
+	switch l := len(split); l {
+	case 1:
+		name = split[0]
+	case 2:
+		prefix = split[0]
+		name = split[1]
+	default:
+		return "", "", fmt.Errorf("invalid count %d of slashes in prefix/name", l)
+	}
+
+	if name == "" {
+		return "", "", errors.New("name cannot be empty")
+	}
+
+	return
+}
+
+func newVM(c *createVM) *v1.VirtualMachine {
+	return &v1.VirtualMachine{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       v1.VirtualMachineGroupVersionKind.Kind,
+			APIVersion: v1.VirtualMachineGroupVersionKind.GroupVersion().String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: v1.VirtualMachineSpec{
+			Running: &c.running,
+			Template: &v1.VirtualMachineInstanceTemplateSpec{
+				Spec: v1.VirtualMachineInstanceSpec{
+					TerminationGracePeriodSeconds: &c.terminationGracePeriod,
+				},
+			},
+		},
+	}
+}
+
+func withRunStrategy(c *createVM, vm *v1.VirtualMachine) error {
+	if _, ok := runStrategies[c.runStrategy]; !ok {
+		return fmt.Errorf("unknown RunStrategy %s", c.runStrategy)
+	}
+
+	runStrategy := v1.VirtualMachineRunStrategy(c.runStrategy)
+	vm.Spec.RunStrategy = &runStrategy
+	vm.Spec.Running = nil
+
+	return nil
+}
+
+func withInstancetype(c *createVM, vm *v1.VirtualMachine) error {
+	kind, name, err := splitPrefixedName(c.instancetype)
+	if err != nil {
+		return err
+	}
+
+	if kind != "" && kind != instancetype.SingularResourceName && kind != instancetype.ClusterSingularResourceName {
+		return fmt.Errorf("invalid instancetype kind %s", kind)
+	}
+
+	vm.Spec.Instancetype = &v1.InstancetypeMatcher{
```

Comment:
I think this could use a comment to outline that we are relying on the `VirtualMachine` mutation webhook to set a default when no Kind is provided and by default that it uses the cluster wide CRD.

Reply:
ACK